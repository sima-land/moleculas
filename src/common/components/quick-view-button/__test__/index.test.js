import React from 'react';
import { mount } from 'enzyme';
import QuickViewButton from '../index';

describe('<QuickViewButton />', () => {
  it('should works without props', () => {
    expect(mount(<QuickViewButton />)).toMatchSnapshot();
  });
  it('should works with default props', () => {
    const spy = jest.fn();
    const wrapper = mount(<QuickViewButton size={15} onClick={spy} className='btn' />);
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
