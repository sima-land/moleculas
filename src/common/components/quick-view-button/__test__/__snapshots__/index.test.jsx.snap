// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuickViewButton /> should works with default props 1`] = `
<ForwardRef
  className="btn"
  onClick={[MockFunction]}
  size={15}
>
  <ForwardRef(SvgQuickView)
    aria-label="Быстрый просмотр"
    className="root btn"
    data-testid="quick-view-button"
    height={24}
    onClick={[MockFunction]}
    role="button"
    size={15}
    width={24}
  >
    <svg
      aria-label="Быстрый просмотр"
      className="root btn"
      data-testid="quick-view-button"
      height={24}
      onClick={[MockFunction]}
      role="button"
      size={15}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m19.94 22.06-3.219-3.217-.339.22A8.956 8.956 0 0 1 11.5 20.5a9 9 0 1 1 9-9c0 1.8-.528 3.476-1.438 4.882l-.22.34 3.219 3.217a1.5 1.5 0 0 1-2.122 2.122ZM9 10h1V9a1.5 1.5 0 1 1 3 0v1h1a1.5 1.5 0 1 1 0 3h-1v1a1.5 1.5 0 1 1-3 0v-1H9a1.5 1.5 0 1 1 0-3Zm2.5-4.5a6 6 0 1 0 0 12 6 6 0 0 0 0-12Z"
      />
    </svg>
  </ForwardRef(SvgQuickView)>
</ForwardRef>
`;

exports[`<QuickViewButton /> should works without props 1`] = `
<ForwardRef>
  <ForwardRef(SvgQuickView)
    aria-label="Быстрый просмотр"
    className="root"
    data-testid="quick-view-button"
    height={24}
    role="button"
    width={24}
  >
    <svg
      aria-label="Быстрый просмотр"
      className="root"
      data-testid="quick-view-button"
      height={24}
      role="button"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m19.94 22.06-3.219-3.217-.339.22A8.956 8.956 0 0 1 11.5 20.5a9 9 0 1 1 9-9c0 1.8-.528 3.476-1.438 4.882l-.22.34 3.219 3.217a1.5 1.5 0 0 1-2.122 2.122ZM9 10h1V9a1.5 1.5 0 1 1 3 0v1h1a1.5 1.5 0 1 1 0 3h-1v1a1.5 1.5 0 1 1-3 0v-1H9a1.5 1.5 0 1 1 0-3Zm2.5-4.5a6 6 0 1 0 0 12 6 6 0 0 0 0-12Z"
      />
    </svg>
  </ForwardRef(SvgQuickView)>
</ForwardRef>
`;
